test/replication-luatest/gh_8121_memtx_mvcc_replication_stream_test.lua

```diff
@@ -0,0 +1,57 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.cluster = cluster:new()
+    cg.master = cg.cluster:build_server({alias = 'master'})
+    cg.replica = cg.cluster:build_server(
+            {
+                alias = 'replica',
+                box_cfg = {
+                    replication = server.build_listen_uri('master'),
+                    memtx_use_mvcc_engine = true,
+                    txn_isolation = 'read-confirmed',
+                }
+            })
+    cg.cluster:add_server(cg.master)
+    cg.cluster:add_server(cg.replica)
+    cg.cluster:start()
+    cg.master:exec(function()
+        local s = box.schema.space.create('s')
+        s:create_index('pk')
+        s:insert{0}
+    end)
+    t.helpers.retrying({}, function()
+        cg.replica:assert_follows_upstream(cg.master:get_instance_id())
+    end)
+end)
+
+g.after_all(function(cg)
+    cg.cluster:drop()
+end)
+
+-- Checks that replication stream does not get transaction conflict errors.
+g.test_replication_stream_transaction_conflict_errors = function(cg)
+    cg.replica:exec(function()
+        box.cfg{replication = ""}
+    end)
+    cg.master:exec(function()
+        box.space.s:delete{0}
+        box.space.s:delete{0}
```

Comment:
I see that the test fails on the current master, which's good.

But wasn't the case with 2 deletes fixed in #8122 ?
For example, when I try to run the original reproducer with 2 deletes, it doesn't cause a conflict.

Reply:
> For example, when I try to run the original reproducer with 2 deletes, it doesn't cause a conflict.

But at the same time you say:
> I see that the test fails on the current master, which's good.

> But wasn't the case with 2 deletes fixed in https://github.com/tarantool/tarantool/issues/8122 ?

No, AFAIC, it wasn't — I run the original reproducer with 2 deletes and it causes conflict on master.
<img width="1223" alt="Screen Shot 2023-01-20 at 11 30 04" src="https://user-images.githubusercontent.com/32464072/213650673-1b83fc8b-6891-48a4-8353-444d47d4a05c.png">